mreg_cli/commands/host_submodules/a_aaaa.py

```diff
@@ -142,6 +143,14 @@ def _ip_remove(name: str, ipaddr: str, ipversion: IP_Version) -> None:
     if not host_ip:
         raise EntityNotFound(f"Host {host} does not have IP {ip}")
 
+    # Check if we fetched the host via a CNAME.
+    # Ensure arg name ends with configured domain
+    # (e.g. "foo" -> "foo.example.com")
+    name_hostname = HostName.parse_or_raise(name)
+    for cname in host.cnames:
+        if not force and name_hostname == cname.name.casefold():
+            raise ForceMissing(f"{name_hostname} is a CNAME for {host.name}, must force.")
+
```

Comment:
This solution isn't great, but it was the best way I could find that avoids additional API calls.

With API calls, we could do this:

```suggestion
    if not force and host.cnames:
        name_hostname = HostName.parse_or_raise(name)
        cname = CNAME.get_by_field("name", name_hostname)
        if cname:
            raise ForceMissing(f"{cname.name} is a CNAME for {host.name}, must force.")
```